### PR TITLE
[Improvement][MasterServer] event response handle parallel 

### PR DESCRIPTION
**Pull request by zwZjut**
## Purpose of the pull request

https://github.com/apache/dolphinscheduler/issues/7450

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

### CODE REVIEW

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/processor/queue/TaskResponsePersistThread.java [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    @Override
+    public void run() {
+        while (!this.events.isEmpty()) {
+            TaskResponseEvent event = this.events.peek();
+            try {
+                persist(event);
+            } catch (Exception e) {
+                logger.error("persist error, task id:{}, instance id:{}", event.getTaskInstanceId(), event.getProcessInstanceId(), e);
+            } finally {
+                this.events.remove(event);
```
Comment (caishunfeng) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/processor/queue/TaskResponsePersistThread.java:76]: if exception happend, should the event be removed?

Comment (zwZjut) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/processor/queue/TaskResponsePersistThread.java:76]: the logic is the same  as before, worker will retry to send event to master

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/processor/queue/TaskResponseService.java [hunk 2]
```diff
@@ -142,80 +174,62 @@ public void run() {
...
+        private void eventHandler() {
+
+            Iterator<Map.Entry<Integer, TaskResponsePersistThread>> iter = processTaskResponseMapper.entrySet().iterator();
+
+            while (iter.hasNext()) {
+                Map.Entry<Integer, TaskResponsePersistThread> entry = iter.next();
+                int processInstanceId = entry.getKey();
+                TaskResponsePersistThread taskResponsePersistThread = entry.getValue();
+                if (taskResponsePersistThread.isEmpty()) {
+                    continue;
```
Comment (caishunfeng) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/processor/queue/TaskResponseService.java:204]: Maybe it's better to add the remove logic here if process instance was finished and no task response event.

**Review by caishunfeng — APPROVED** [review]
+1

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/processor/queue/TaskResponsePersistThread.java [hunk 3]
```diff
@@ -0,0 +1,172 @@
+package org.apache.dolphinscheduler.server.master.processor.queue;
```
Comment (caishunfeng) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/processor/queue/TaskResponsePersistThread.java:18]: add license header

Comment (zwZjut) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/processor/queue/TaskResponsePersistThread.java:18]: fixed

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/processor/queue/TaskResponseService.java [hunk 4]
```diff
@@ -66,13 +66,32 @@
...
+    private final ConcurrentHashMap<String, TaskResponsePersistThread> eventHandlerMap = new ConcurrentHashMap<>();
+
+    private ListeningExecutorService listeningExecutorService;
+
+    private ExecutorService eventExecService;
+
+    /**
+     * task response mapper
+     */
+    private final ConcurrentHashMap<Integer, TaskResponsePersistThread> processTaskResponseMapper = new ConcurrentHashMap<>();
```
Comment (geosmart) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/processor/queue/TaskResponseService.java]: the name make me  think the map was a DAO object.

maybe change it to `processTaskResponseMap`

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/processor/queue/TaskResponseService.java [hunk 5]
```diff
@@ -142,84 +175,72 @@ public void run() {
...
     /**
-     * persist  taskResponseEvent
-     *
-     * @param taskResponseEvent taskResponseEvent
+     * event handler thread
      */
-    private void persist(TaskResponseEvent taskResponseEvent) {
-        Event event = taskResponseEvent.getEvent();
-        Channel channel = taskResponseEvent.getChannel();
+    class EventHandler extends Thread {
```
Comment (geosmart) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/processor/queue/TaskResponseService.java]: name can be more concrete like `TaskResponseEventHandler`

and the comment `event handler thread` say nothing, just remove it or describe what the thread do

**Review by geosmart — COMMENTED** [review]
add  naming review

**Pull request merged**